### PR TITLE
Enable react/prop-types eslint rule; add missing propTypes

### PR DESCRIPTION
**Pull request by benvinegar**
This adds an eslint rule requiring everyone to specify `propTypes` for each React component.

Specifying prop types adds run-time prop validation that helps locate and eliminate bugs. This checking is only done in development; the code is stripped in production builds.

We were really inconsistent in our use of specifying them, so adding this rule will make sure we always specify prop types going forward.

### CODE REVIEW

**Comment by mattrobenolt** [comment]
> This branch has conflicts that must be resolved

glhf rebasing

File: src/sentry/static/sentry/app/components/selectInput.jsx [hunk 1]
```diff
@@ -2,6 +2,15 @@ import React from 'react';
 import jQuery from 'jquery';
 
 const SelectInput = React.createClass({
+  propTypes: {
+    disabled: React.PropTypes.bool,
+    multiple: React.PropTypes.bool,
+    required: React.PropTypes.bool,
+    placeholder: React.PropTypes.string,
+    value: React.PropTypes.string,
+    onChange: React.PropTypes.func,
```
Comment (mattrobenolt) [inline src/sentry/static/sentry/app/components/selectInput.jsx]: trailing comma or no trailing comma? Not sure if you had a rule here yet or a preference. Seems that you generally don't have one.

**Comment by benvinegar** [comment]
> glhf rebasing

ok

```bash
(sentry)➜  sentry git:(bb87d80) ✗ git status
rebase in progress; onto bb87d80
You are currently rebasing branch 'proptypes' on 'bb87d80'.
...

Unmerged paths:
  (use "git reset HEAD <file>..." to unstage)
  (use "git add <file>..." to mark resolution)

	both modified:   src/sentry/static/sentry/app/components/todos.jsx
```

**Comment by mattrobenolt** [comment]
:+1:

File: src/sentry/static/sentry/app/components/selectInput.jsx (hunk 1 above)
Comment (benvinegar) [inline src/sentry/static/sentry/app/components/selectInput.jsx]: > Seems that you generally don't have one.

I'd prefer not to have trailing comma. It's not common practice in JavaScript like it is in Python.

**Pull request merged**